src/sdk/actions.ts

```diff
@@ -130,3 +142,47 @@ async function returnDataToKernel(repoToken: string, stateId: string, output: ob
     },
   });
 }
+
+function createStructuredMetadata(className: string | undefined, logReturn: LogReturn) {
+  let logMessage, metadata;
+  if (logReturn) {
+    logMessage = logReturn.logMessage;
+    metadata = logReturn.metadata;
+  }
+
+  const jsonPretty = sanitizeMetadata(metadata);
+  const stackLine = new Error().stack?.split("\n")[2] ?? "";
+  const caller = stackLine.match(/at (\S+)/)?.[1] ?? "";
+  const ubiquityMetadataHeader = `<!-- ${HEADER_NAME} - ${className} - ${caller} - ${metadata?.revision}`;
```

Comment:
we don't use `metadata.revision` really anymore we could remove it or find a way to support it. It requires that the plugin or sdk pull in the current sha of the `development` branch as that's what it's original purpose was.

Reply:
The revision is immensely valuable for debugging and logging. We should make this high priority to fix.

Reply:
Revision should point to the current version of the plugin. Within actions this is easy to get, within workers I don't think this is possible, instead we could probably use the version id?
![Uploading image.png…]()

Reply:
Either way this problem is a concern for the logger not the SDK, will open a ticket for it.